enterprise/backend/src/metabase_enterprise/transforms/api.clj

```diff
@@ -245,6 +238,20 @@
       (transforms.canceling/cancel-run! (:id run))))
   nil)
 
+(api.macros/defendpoint :post "/:id/flush-watermark"
+  "Flush the watermark for an incremental transform with keyset strategy.
+  This will cause the next run to recompute the watermark from scratch."
+  [{:keys [id]} :- [:map
+                    [:id ms/PositiveInt]]]
+  (log/info "flushing watermark for transform" id)
+  (api/check-superuser)
+  (let [transform (api/check-404 (t2/select-one :model/Transform id))]
+    (api/check-400 (= :keyset (some-> transform :source :source-incremental-strategy :type keyword))
```

Comment:
we we do this `(= :keyset (some-> transform :source :source-incremental-strategy :type keyword))` in a few places, we should have a util function for it